src/clangd-context.ts

```diff
@@ -206,4 +212,65 @@ export class ClangdContext implements vscode.Disposable {
       this.client.stop();
     this.subscriptions = []
   }
+
+  private async remapDefinitionSymlink(
+    document: vscode.TextDocument, position: vscode.Position,
+    token: vscode.CancellationToken,
+    next: vscodelc.ProvideDeclarationSignature) {
+  let definition = await next(document, position, token);
+  if (!config.get<boolean>('followSymlinks')) {
+    return definition;
+  }
+  if (definition) {
+    if (Array.isArray(definition)) {
+      const res = await Promise.all(definition.map(async (e) => {
+        if ('targetUri' in e)
+          return e;
+        else {
+          if (e.uri.scheme != 'file')
+            return e;
+
+          const s =
+              await vscode.workspace.fs.stat(vscode.Uri.file(e.uri.path));
+          if (s.type & vscode.FileType.SymbolicLink) {
+            const p = await this.safeFollowSymLink(e.uri.fsPath)
+            return new vscode.Location(vscode.Uri.file(p), e.range)
+          }
+
+          return e;
+        }
+      })) as vscode.Declaration
+
+      return res;
+    } else {
+      if (definition.uri.scheme != 'file')
```

Comment:
It would have been preferable to use the vscode filesystem api here, but while it has functionality for *checking* if a file is a symlink, it doesn't actually have a function to follow it. Gating behind a file uri scheme should avoid situations where you've got some remote filesystem open and we can't actually follow the symlinks